### PR TITLE
[Feature Store] Fix `_ingest_with_spark` does not raise proper errors

### PR DESCRIPTION
**Pull request by davesh0812**
[ML-3298](https://jira.iguazeng.com/browse/ML-3298)

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -840,6 +841,8 @@ def _ingest_with_spark(
             df = source.filter_df_start_end_time(df, timestamp_key)
         if featureset.spec.graph and featureset.spec.graph.steps:
             df = run_spark_graph(df, featureset, namespace, spark)
+        if isinstance(df, Response) and df.status_code == 400:
```
Comment (assaf758) [inline mlrun/feature_store/api.py:844]: why only 400?

File: mlrun/feature_store/api.py [hunk 2]
```diff
@@ -840,6 +841,8 @@ def _ingest_with_spark(
...
+            mlrun.errors.raise_for_status_code(df.status_code, df.body.split(": ")[1])
```
Comment (assaf758) [inline mlrun/feature_store/api.py:845]: what is the split()[1] extracting? would it always work?

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (assaf758) [inline mlrun/feature_store/api.py:844]: please add extra line of space above

**Review by assaf758 — COMMENTED** [review]
please add to the RCA comment - what error was injected by the test?

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (davesh0812) [reply mlrun/feature_store/api.py:844]: it is always 400 you can see it on server.run func , line 268 in server.py

File: mlrun/feature_store/api.py (hunk 2 above)
Comment (davesh0812) [reply mlrun/feature_store/api.py:845]: Also because server.run behavior

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (davesh0812) [reply mlrun/feature_store/api.py]: I changed it to be != 0

File: mlrun/feature_store/api.py (hunk 2 above)
Comment (davesh0812) [reply mlrun/feature_store/api.py:846]: Also result of server.run fun behavior

**Review by assaf758 — APPROVED** [review]

**Pull request merged**